src/main/kotlin/com/terraformation/backend/api/OpenApiConfig.kt

```diff
@@ -142,10 +144,16 @@ class OpenApiConfig(private val keycloakInfo: KeycloakInfo) : OpenApiCustomiser
             propertySchema.description = propertyAnnotation.description
           }
 
+          // ArbitraryJsonObject should always allow additional properties.
+          if (propertySchema is ObjectSchema &&
+              property.returnType.isSubtypeOf(typeOf<ArbitraryJsonObject?>())) {
+            propertySchema.additionalProperties = true
+          }
+
           // Map<*,*> fields default to additionalProperties values that say the values are all
           // objects, which is wrong; values could be other JSON types too.
           if (propertySchema is ArraySchema && propertySchema.items is MapSchema) {
-            propertySchema.items.additionalProperties = null
+            propertySchema.items.additionalProperties = true
```

Comment:
Is the comment above this block still accurate?

Reply:
The comment is more accurate now than it was before. There are three possibilities for the `additionalProperties` option:

* If it's missing or null, the schema says you can't include properties other than the ones explicitly listed in the schema. That's what the previous version of this code was doing by setting it to null. Our TypeScript code doesn't care if an object has fields that aren't in the interface, but if we'd, say, generated a Java client from this schema, it would have had no way to add arbitrary key/value pairs.
* If it's `true`, you can include arbitrary properties and their values can be of any type.
* If it's an OpenAPI type (a child object with a schema definition, a reference to another type defined in the schema, or a built-in type name like `object` or `string`) it means you can include arbitrary properties but their values all have to be of the specified type. This is the default behavior (it uses `object` as the type).

https://swagger.io/docs/specification/data-models/dictionaries/ has more gory detail.

Reply:
ah, gotcha. Thanks for explaining!